data/wp/wp-content/plugins/epfl/shortcodes/epfl-map/epfl-map.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/**
+ * Plugin Name: EPFL Map shortcode
+ * Description: provides a shortcode to display a map from map.epfl.ch
+ * @version: 1.1
+ * @copyright: Copyright (c) 2017 Ecole Polytechnique Federale de Lausanne, Switzerland
+ */
+
+/**
+ * Check the parameters
+ *
+ * Return True if all parameters are populated
+ *
+ * @param $query: query example: the office, the person
+ * @param $lang: language
+ * @return True if all parameters are populated
+ */
+function epfl_map_check_parameters( string $query, string $lang): bool
+{
+    return $query !== "" && $lang !== "";
+}
+
+/**
+ * Execute the shortcode
+ *
+ * @attributes: array of all input parameters
+ * @content: the content of the shortcode. In our case the content is empty
+ * @return html of shortcode
+ */
+function epfl_map_process_shortcode( $attributes, string $content = null ): string
+{
+    // get parameters
+    $atts = shortcode_atts(array(
+        'query'  => '',
+        'lang'   => '',
+    ), $attributes);
+
+    // sanitize parameters
+    $query  = sanitize_text_field($atts['query']);
+    $lang   = sanitize_text_field($atts['lang']);
+
+    // check parameters
+    if ( false == epfl_map_check_parameters($query, $lang) ) {
+        return "";
```

Comment:
Tu veux pas passer par le mécanisme "globale" au plugin "all-in-one" pour l'affichage des erreurs?  même si on n'a pas encore défini comment on affichait.

Reply:
Ajouté !